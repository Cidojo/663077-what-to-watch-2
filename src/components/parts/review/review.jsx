import * as React from 'react';
import * as PropTypes from 'prop-types';

import {formatDate} from './../../../utils/utils.js';

const Review = (props) => {
  const {review} = props;

  const formattedDate = formatDate(new Date(review.date));

  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{review.comment}</p>

        <footer className="review__details">
          <cite className="review__author">{review.user.name}</cite>
          <time
            className="review__date"
            dateTime={formattedDate.dateTime}
          >
            {formattedDate.verbose}
          </time>
        </footer>
      </blockquote>

      <div className="review__rating">{review.rating}</div>
    </div>
  );
};

Review.propTypes = {
  review: PropTypes.shape({
    id: PropTypes.number,
    user: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
    }),
    rating: PropTypes.number,
    comment: PropTypes.string,
    date: PropTypes.string
  })
};

Review.defaultProps = {
  review: {
    id: 0,
    user: {
      id: 0,
      name: ``,
    },
    rating: 0,
    comment: ``,
    date: ``
  }
};

export {Review};
