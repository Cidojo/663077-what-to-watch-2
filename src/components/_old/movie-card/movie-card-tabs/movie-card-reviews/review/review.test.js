import * as React from 'react';
import {Review} from './review.jsx';
import * as renderer from 'react-test-renderer';

it(`should render component without errors`, () => {
  const review = {
    id: 0,
    user: {
      id: 0,
      name: ``,
    },
    rating: 0,
    comment: ``,
    date: ``
  };

  const tree = renderer
    .create(<Review
      review={review}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
