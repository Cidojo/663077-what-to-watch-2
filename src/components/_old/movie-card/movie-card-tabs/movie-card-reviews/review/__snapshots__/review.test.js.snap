// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component without errors 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        
      </cite>
      <time
        className="review__date"
        dateTime="NaN-NaN-NaN"
      >
        undefined NaN, NaN
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    0
  </div>
</div>
`;
